torch/testing/_internal/common_utils.py

```diff
@@ -5312,3 +5312,36 @@ def repl_frame(m):
         s = re.sub(r"Cannot export model.+\n\n", "", s)
     s = re.sub(r" +$", "", s, flags=re.MULTILINE)
     return s
+
+def recover_orig_fp32_precision():
```

Comment:
Don't we expect test to always run in the default precision setting by default? So we don't need to stash everything and we can just restore everything to the default value on exit?